<?php

namespace Feature\Api\Company;

use App\Models\Company;
use App\Models\User;
use Tests\TestCase;

class StoreCompanyTest extends TestCase
{
    private User $user;

    private Company $company;

    protected function setUp(): void
    {
        parent::setUp(); // TODO: Change the autogenerated stub

        $this->user = User::factory()->create();
    }

    /**
     * A basic feature test example.
     */
    public function test_store_company(): void
    {
        // sail artisan test --filter=StoreCompanyTest

        $this->company = Company::factory()->make();
        $this->company->spokesperson_id = $this->user->id;

        // вот эта штука покажет дополнительные ошибки!
        $this->withoutExceptionHandling();

        $response = $this->actingAs($this->user)
            ->post(route('company.store'), $this->company->getAttributes())
            //->assertSessionHasNoErrors() // эта штука также выдает ошибку!
        ;

        // это выдает 422 ошибку, но непонятно где именно ошибка
        //$response = $this->actingAs($this->user)
        //    ->json('POST', route('post.store'), $this->post->getAttributes());

        //$response->dump();
        $response->assertCreated();

        $response->assertJsonIsObject();
    }
}
